File: src/types/workspace.ts
export interface Workspace {
  id: string;
  repository_id: string;
  name: string;
  branch: string;
  worktree_path: string | null;
  status: "active" | "archived";
  created_at: string;
  updated_at: string;
}

export interface CreateWorkspaceRequest {
  repository_id: string;
  name: string;
  branch: string;
  base_branch?: string;
}

File: src-tauri/src/model/workspace.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Workspace {
    pub id: String,
    pub repository_id: String,
    pub name: String,
    pub branch: String,
    pub worktree_path: Option<String>,
    pub status: String,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Clone, Deserialize)]
pub struct CreateWorkspaceRequest {
    pub repository_id: String,
    pub name: String,
    pub branch: String,
    pub base_branch: Option<String>,
}
